src/main/java/seedu/address/logic/commands/AddNoteCommand.java

```diff
@@ -0,0 +1,110 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_IC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NOTE;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.IdentityCardNumberMatchesPredicate;
+import seedu.address.model.person.Note;
+import seedu.address.model.person.Person;
+
+/**
+ * Changes the note of an existing person in the address book.
+ */
+public class AddNoteCommand extends Command {
+
+    public static final String COMMAND_WORD = "addnote";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the note of the person whose profile matches "
+            + "the specified IC (case-insensitive). "
+            + "Existing remark will be appended by default. To replace the original note, add -replace at "
+            + "the end of your command. E.g. addnote i/S0123456Q n/Diabetes -replace\n"
+            + "Parameters: "
+            + "[" + PREFIX_IC + "IC] "
+            + "[" + PREFIX_NOTE + "NOTE] \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_IC + " S0123456Q "
+            + PREFIX_NOTE + "Healthy.";
+
+
+    public static final String MESSAGE_ADD_NOTE_SUCCESS = "Added note to Person: %1$s";
+    public static final String MESSAGE_DELETE_NOTE_SUCCESS = "Removed note from Person: %1$s";
+    private final IdentityCardNumberMatchesPredicate ic;
```

Comment:
Could we change `ic` to `icPredicate` or `predicate` or something like that instead? I think it would be a bit better for readability.